code/modules/admin/topic.dm

```diff
@@ -902,14 +902,15 @@
 	else if(href_list["boot2"])
 		var/mob/M = locateUID(href_list["boot2"])
 		if(ismob(M))
-			if(M.client && M.client.holder && (M.client.holder.rights & R_BAN))
-				to_chat(usr, "<span class='warning'>[key_name_admin(M)] cannot be kicked from the server.</span>")
-				return
-			to_chat(M, "<span class='warning'>You have been kicked from the server</span>")
-			log_admin("[key_name(usr)] booted [key_name(M)].")
-			message_admins("<span class='notice'>[key_name_admin(usr)] booted [key_name_admin(M)].</span>", 1)
-			//M.client = null
-			del(M.client)
+			if(alert("Kick [M.ckey]?",,"Yes","No") == "Yes")
```

Comment:
Note that this introduces a delay here - meaning that the mob could null out (gets deleted in some way) or mob's client could either null out (ghosting) or be swapped (mindswap) before the admin can click yes. It might be better to grab a reference to the client/ckey before starting the check.

Reply:
Updated it, it now gets a reference to the client first, and makes sure it's not null (turns out things like already kicked players and aghosted admins can have ckeys, but no clients).